src/calculationEngine/entitys/CeEntity.java

```diff
@@ -70,15 +95,16 @@ private int rollSalt() {
             }
         }
         for (int i = 0; i < pickedAttacks.length; i++) {
-            pickedAttacks[i] = new Attack(typedAttacks.get(random.nextInt(typedAttacks.size())));
+            int index = random.nextInt(typedAttacks.size());
+            pickedAttacks[i] = new Attack(typedAttacks.get(index));
+            typedAttacks.remove(index);
```

Comment:
You don't need this.

Reply:
You 100 % Sure?

Reply:
https://www.tutorialspoint.com/java/util/arraylist_remove.htm